### ISSUE TITLE
Error at Startup

### ISSUE THREAD

**Issue by bacco007** [bremor/bureau_of_meteorology#63]
I'm getting the following errors when Home Assistant starts up - I dont appear to get it at any other time

```yaml
2021-01-02 09:22:37 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform bureau_of_meteorology
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 316, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 530, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 101, in async_added_to_hass
    self._update_callback()
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 105, in _update_callback
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 296, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 323, in _async_write_ha_state
    attr.update(self.device_state_attributes or {})
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 180, in device_state_attributes
    attr[ATTR_DATE] = self.collector.daily_forecasts_data["data"][self.day]["date"]
IndexError: list index out of range
2021-01-02 09:22:37 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up bureau_of_meteorology platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 207, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 316, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 530, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 101, in async_added_to_hass
    self._update_callback()
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 105, in _update_callback
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 296, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 323, in _async_write_ha_state
    attr.update(self.device_state_attributes or {})
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 180, in device_state_attributes
    attr[ATTR_DATE] = self.collector.daily_forecasts_data["data"][self.day]["date"]
IndexError: list index out of range
```

**Comment by ausmkv** [comment]
@bacco007 any chance you entered the forecast days as "7" during setup? I was having exactly the same issue with the forecast range set to 7. When I deleted and re-added the integration, I noticed that the scroller range on forecast days is 0 - 5 rather than 0 - 7 and then read the instructions that say (day 5 is always available, day 6 and 7 are only sometimes available).... 🤦‍♂️ 
With forecast days set to 5, I no longer have the errors above.
![image](https://user-images.githubusercontent.com/643769/103450579-b5cd2980-4d0c-11eb-9f7f-aaa1fc8d96ce.png)
